wbooks-api/spec/controllers/books_controller_spec.rb

```diff
@@ -1,18 +1,92 @@
 require 'rails_helper'
 
 RSpec.describe BooksController, type: :controller do
+  include_context 'Authenticated User'
 
-  describe "GET #index" do
-    it "returns http success" do
-      get :index
-      expect(response).to have_http_status(:success)
+  describe 'GET #index' do
+    context 'When fetching all the books' do
+      let!(:books) { create_list(:book, 4) }
+      before do
+        get :index
+      end
+ 
+      it 'responds with the lists of books' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          books, serializer: BookIndexSerializer
+        ).to_json
+        expect(response_body['page'].to_json).to eq expected
+      end
+ 
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'responds with all the books' do
+        expected = books.count
+        expect(response_body['page'].length).to eq expected
+      end
+    end
+
+    context 'When given a limit of records expected' do
+      let!(:books) { create_list(:book, 5) }
+
+      it 'responds with limit of records' do
+        get :index, params: { limit: 3 }
+        expect(response_body['page'].length).to eq (3)
+      end
+
+      it 'responds with number of next page' do
+        get :index, params: { limit: 3 }
+        expect(response_body['next_page']).to_not eq (nil) 
+      end
+
+      it 'responds with next_page null' do
+        get :index, params: { limit: 5 }
+        expect(response_body['next_page']).to eq (nil) 
+      end
     end
   end
 
-  describe "GET #show" do
-    it "returns http success" do
-      get :show
-      expect(response).to have_http_status(:success)
+  describe 'GET #show' do
+    context 'When given a correct ID' do
+      let!(:book) { create(:book) }
+      before do
+        get :show, params: { id: book.id }
+      end
+
+      it 'responds with a book json' do
+        expect(response_body.to_json).to eq BookShowSerializer.new(
+          book
+        ).to_json
+      end
+ 
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'responds with expected attributes for book' do
+        expect(response_body['genre']).to eq book.genre
+        expect(response_body['author']).to eq book.author
+        expect(response_body['title']).to eq book.title
+        expect(response_body['year']).to eq book.year
+        expect(response_body['publisher']).to eq book.publisher
+        expect(response_body['image_url']).to eq book.image
+      end
+    end
+
+    context 'When given a non-existent ID' do
+      let!(:error) { { "error": "Couldn't find Book with 'id'=1" } }
+      before do
+        get :show, params: { id: 1 }
+      end
+
+      it 'responds with an error message' do
+        expect(response_body.to_json).to eq error.to_json
```

Comment:
It's not a good idea to check for a message string (for example, the message could change while the error would be the same semantically and it'd make this fail) unless you can reference it without copy-pasting the full string (for example, using a i18n string or a constant). A good alternative here is to check that the error key exists and that the HTTP code is the one you expected. 

(For non-controller methods, an alternative would be to use this this matcher:
https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/raise-error-matcher)